torch/distributed/__init__.py

```diff
@@ -37,8 +37,17 @@ def destroy_process_group():
     Destroy the initialized distributed package
     """
     global _backend
+    global _initialized
+    torch._C._dist_destroy_process_group()
     _backend = dist_backend.UNDEFINED
-    return torch._C._dist_destroy_process_group()
+    _initialized = 0
```

Comment:
do we even need this flag? We could treat `_backend is `dist_backend.UNDEFINED`  as `not _initialized`

Reply:
_Initialized has either the value of _INITIALIZED_PG or _INITIALIZED_MW and has been used in a lot of the places in the file. Let's keep it for now.  Removed unnecessary global and rename the function.